modules/custom-status/custom-status.php

```diff
@@ -441,13 +487,9 @@ public function remove_or_add_publish_capability_for_user( $allcaps, $cap, $args
 
 		$status_before_publish = $custom_statuses[ array_key_last( $custom_statuses ) ];
 
-		// If the post status is not the last status, remove the publish capability or else add it back in
+		// Ensure publishing is disabled for all but the last status
 		if ( $status_before_publish->slug !== $post->post_status ) {
-			// Remove the publish capability based on the post type
 			$allcaps[ $supported_publish_caps_map[ $post->post_type ] ] = false;
-		} else {
-			// Remove the publish capability based on the post type
-			$allcaps[ $supported_publish_caps_map[ $post->post_type ] ] = true;
```

Comment:
I noticed that we were explicitly setting a capability to `true` here, and it may be in error. If this status doesn't match an expected one, we should leave capabilities as default without modifying them.